In [20]:
# Mithil Patel 
# DSC 530 Week 8 Assignment
# 31th July, 2022

# Importing libraries
import thinkstats2
import numpy as np
import nsfg
import first
import pandas as pd

# Importing dataset and then righ joining with other dataset
live, firsts, others = first.MakeFrames()
live = live[live.prglngth>30]
resp = nsfg.ReadFemResp()
resp.index = resp.caseid
join = live.join(resp, on='caseid', rsuffix='_r')

Problem 11.1

There are multiple variables we can use in order to estimate a co-worker's pregnancy length. Several variables such as birth weight will not be useful, and therefore should not be included in your model. In our model, we should use the mother's age, race, birth order, multiple births (nbrnaliv), baby's sex, and income variables to predict the baby's date of birth.

Problem 11.3

In [27]:
# Creating a poisson model 
formula = 'numbabes ~ age_r + C(race) + totincr + educat'
poission_model = smf.poisson(formula, data=join).fit()

# Making predictions
user_variable = ['age_r', 'race', 'totincr', 'educat']
pred = pd.DataFrame([[35, 1, 14, 16]], columns= user_variable)
poission_model.predict(pred)

Optimization terminated successfully.
         Current function value: 1.687055
         Iterations 5


0    2.342182
dtype: float64

Based on the poission constructed model above, we can predict that a woman who is 35 years old, black, and a college graduate whose annual household income exceeds $75,000 will have approximately 3 children.

Problem 11.4

In [36]:
## Creating a multinomial logistic regression model 
log_formula = 'rmarital ~ age_r + C(race) + totincr + educat'
multi_log = smf.mnlogit(log_formula, data= join)
log_res = multi_log.fit()

# Making predictions
log_variable = ['age_r', 'race', 'totincr', 'educat']
log_pred = pd.DataFrame([[25, 2, 9, 12]], columns= log_variable)
log_res.predict(log_pred)

Optimization terminated successfully.
         Current function value: 1.087603
         Iterations 8


0         1        2         3         4         5
0  0.662055  0.144018  0.00187  0.049168  0.037801  0.105088

Using the multinomial logistic regression, a woman who is 25 years old, white, and a high school graduate whose annual household income is about $45,000 is approximately 66% likely to be married while 14% chance of cohabitating.